File: p003/p003.js
var p003 = function (number) {
	'use strict';

	if (number === 0) return Infinity;

	var i = 2,
		ceiling = Math.sqrt(Math.abs(number));
	while (i < ceiling) {
		for (i = 2; i <= ceiling; i += 1) {
			if (number % i === 0) {
				number /= i;
				break;
			}
		}
		ceiling = parseInt(Math.sqrt(Math.abs(number)));
	}
	return number;
}

if (arguments.length === 1) {
	println(p003(parseInt(arguments[0])));
} else {
	println('Usage: p003 <integer>');
	exit(1);
}

File: p003/p003.py
import math as _math
import sys as _sys

def p003(number):
	if number == 0: return float('inf')

	factored = False
	while True:
		for i in range(2, _math.floor(_math.sqrt(abs(number))) + 1):
			if number % i == 0:
				number /= i
				break
		else:
			return int(number)

if __name__ == '__main__':
	if len(_sys.argv) == 2:
		print(p003(int(_sys.argv[1])))
	else:
		print('Usage: p003 <integer>')
		_sys.exit(1)
